plugins/librato/measurement.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2017 Mesosphere, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/dcos/dcos-metrics/plugins"
+)
+
+// measurement represents each point in a time series
+type measurement struct {
+	Name  string            `json:"name"`
+	Value float64           `json:"value"`
+	Tags  map[string]string `json:"tags"`
+	Time  int64             `json:"time"`
+}
+
+func newMeasurement() *measurement {
+	return &measurement{
+		Tags: make(map[string]string),
+	}
+}
+
+func (m *measurement) floorTime(interval int64) {
+	m.Time = m.Time / interval * interval
+}
+
+func (m *measurement) String() string {
+	return m.Name
+}
+
+func (m *measurement) setValue(value interface{}) error {
+	val, err := plugin.DatapointValueToFloat64(value)
+	if err != nil {
+		return fmt.Errorf("Could not set value: %v", err)
+	}
+	m.Value = val
+	return nil
+}
+
+func (m *measurement) addTag(name string, value string) error {
+	matched, err := regexp.MatchString(`^[-.:_\w]+\z{1,64}$`, name)
+	if err != nil {
+		return fmt.Errorf("Error occurred matching tag name: %v", err)
```

Comment:
you can use `%s` for err, since it implements Stringer interface. %v is more expensive